### PR TITLE
Offline simulation support in XEB fidelity estimator

### PR DESCRIPTION
**Pull request by viathor**

### CODE REVIEW

File: cirq/experiments/fidelity_estimation.py [hunk 1]
```diff
@@ -81,6 +85,11 @@ def linear_xeb_fidelity(
...
-    output_state = final_wavefunction(circuit, qubit_order=qubit_order)
-    output_probabilities = np.abs(output_state)**2
-    return dim * np.mean(output_probabilities[bitstrings]) - 1
+    if amplitudes is None:
+        output_state = final_wavefunction(circuit, qubit_order=qubit_order)
+        output_probabilities = np.abs(output_state)**2
+        bitstring_probabilities = output_probabilities[bitstrings]
+    else:
+        bitstring_probabilities = np.array(
+            [np.abs(amplitudes[bitstring])**2 for bitstring in bitstrings])
```
Comment (kevinsung) [inline cirq/experiments/fidelity_estimation.py]:
```
bitstring_probabilities = np.abs([amplitudes[bitstring] for bitstring in bitstrings])**2
```

File: cirq/experiments/fidelity_estimation.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def linear_xeb_fidelity(
         circuit: Circuit,
         bitstrings: Sequence[int],
         qubit_order: QubitOrderOrList = QubitOrder.DEFAULT,
+        amplitudes: Optional[Dict[int, complex]] = None,
```
Comment (kevinsung) [inline cirq/experiments/fidelity_estimation.py]: Mapping instead of Dict?

File: cirq/experiments/fidelity_estimation.py (hunk 1 above)
Comment (viathor) [reply cirq/experiments/fidelity_estimation.py]: Good catch! Thanks!

File: cirq/experiments/fidelity_estimation.py (hunk 2 above)
Comment (viathor) [reply cirq/experiments/fidelity_estimation.py]: Done.

**Pull request merged**